Added records by category

polar1s505 opened the pull request:
Implemented an option to get records by specific category

polar1s505 committed:
added records by category

Implemented an option to get records by specific category

polar1s505 committed:
minor fix

Puchkof commented on FinanceTracker/FinanceTracker/Commands/GetRecordsByCategory.cs:
```diff
@@ -0,0 +1,18 @@
+
+namespace FinanceTracker.Commands
+{
+    public class GetRecordsByCategory : ICommand
```
Maybe its time to add interface for Commands that returns results? Like ICommand<T> with method T Execute<T>

polar1s505 committed:
Refactor command logic

1) Made ICommand generic
2) Add DisplayCommandInfo class

P.S. Nevermind "Statistics" classes, this funcionality in progress

polar1s505 committed:
implemented statistic func

1) Added statistic for month
2) Added statistic for week

Puchkof commented on FinanceTracker/FinanceTracker/Commands/GetStatisticsBase.cs:
```diff
@@ -0,0 +1,35 @@
+            var operations = account.Operations
+                .Where(operation => operation.Date >= startDate && operation.Date <= endDate);
+
+            foreach (var operation in operations)
```
I think you can use Aggregate here https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.aggregate?view=net-8.0

Puchkof commented on FinanceTracker/FinanceTracker/Program.cs:
```diff
@@ -21,20 +21,32 @@
-            command = new AddExpenceCommand();
-            command.Execute(account);
+            ICommand<Operation> addExpenceCommand = new AddExpenceCommand();
+            account.UpdateBalance(addExpenceCommand.Execute(account));
```
I think this should be executed inside command and you need to return result of this operation

Puchkof commented on FinanceTracker/FinanceTracker/Program.cs:
```diff
@@ -21,20 +21,32 @@
-            command = new AddIncomeCommand();
-            command.Execute(account);
+            ICommand<Operation> addIncomeCommand = new AddIncomeCommand();
+            account.UpdateBalance(addIncomeCommand.Execute(account));
```
I think this should be executed inside command and you need to return result of this operation

polar1s505 committed:
use Aggregate for statistic class, changed add commands

1) Used Aggregate instead of foreach loop
2) Change Execute method in AddOperationBase class
